core/modules/block/block.module

```diff
@@ -237,6 +297,11 @@ function block_custom_block_form($edit = array(), $stand_alone = TRUE) {
       '#default_value' => TRUE,
       '#disabled' => TRUE,
     );
+
+    // Add a notice for translation.
+    if (module_exists('language')) {
+      $form['reusable']['#description'] .= t('NOTE: You have module "language" enabled. If you want to make this block translatable, this block must be marked as reusable.');
```

Comment:
I would suggest simplifying this down to "If you want to make this block translatable, this block must be marked as reusable."